pennylane/pauli/conversion.py

```diff
@@ -14,13 +14,133 @@
 """
 Utility functions to convert between ``~.PauliSentence`` and other PennyLane operators.
 """
+from typing import Union
+from operator import matmul
+from itertools import product
 from functools import reduce, singledispatch
 
+import numpy as np
+
 from pennylane.operation import Tensor
 from pennylane.ops import Hamiltonian, Identity, PauliX, PauliY, PauliZ, Prod, SProd, Sum
 
-from .pauli_arithmetic import PauliWord, PauliSentence, X, Y, Z
 from .utils import is_pauli_word
+from .pauli_arithmetic import PauliWord, PauliSentence, I, X, Y, Z, mat_map, op_map
+
+
+def pauli_decompose(
+    H, hide_identity=False, wire_order=None, pauli=False
```

Comment:
Not sure how much thought and discussion already went into discussing the naming. Personally, I find just `pauli` a bit ambiguous. How would you feel about `return_pauli` or something similarly more descriptive instead?

Reply:
or `return_sentence`

Reply:
I think we are locked in for the name, but thanks for the suggestions haha